src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,126 @@
+            createManufacturerStatement.setString(2, manufacturer.getCountry());
+            createManufacturerStatement.setLong(3, manufacturer.getId());
+            createManufacturerStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't update manufacturer: " + manufacturer, e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        checkId(id);
+        try (Connection connection = ConnectionUtil.getConnection();
+                Statement getAllManufacturerStatement = connection.createStatement()) {
+            ResultSet resultSet = getAllManufacturerStatement
+                    .executeQuery("SELECT * FROM manufacturers WHERE is_deleted = false");
+            while (resultSet.next()) {
+                Long idCurrent = resultSet.getObject("id", Long.class);
+                if (idCurrent.equals(id)) {
+                    String name = resultSet.getString("name");
+                    String country = resultSet.getString("country");
+                    Manufacturer manufacturer = new Manufacturer();
+                    manufacturer.setName(name);
+                    manufacturer.setCountry(country);
+                    manufacturer.setId(idCurrent);
+                    return Optional.of(manufacturer);
+                }
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't get manufacturer by ID " + id + " from DB", e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        List<Manufacturer> allManufacturers = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                Statement getAllManufacturerStatement = connection.createStatement()) {
+            ResultSet resultSet = getAllManufacturerStatement
+                    .executeQuery("SELECT * FROM manufacturers WHERE is_deleted = false");
+            while (resultSet.next()) {
+                Long id = resultSet.getObject("id", Long.class);
+                String name = resultSet.getString("name");
+                String country = resultSet.getString("country");
+                Manufacturer manufacturer = new Manufacturer();
+                manufacturer.setName(name);
+                manufacturer.setCountry(country);
+                manufacturer.setId(id);
+                allManufacturers.add(manufacturer);
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't get all manufacturers from DB", e);
+        }
+        return allManufacturers;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        checkId(id);
+        String deleteFormat = "UPDATE manufacturers SET is_deleted = true WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createManufacturerStatement =
```

Comment:
```suggestion
    public boolean delete(Long id) {
        String deleteFormat = "UPDATE manufacturers SET is_deleted = true WHERE id = ?";
        try (Connection connection = ConnectionUtil.getConnection();
                PreparedStatement createManufacturerStatement =
```
this check is redundant